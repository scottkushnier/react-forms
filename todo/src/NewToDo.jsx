
import { useState } from 'react';
  
const initialFormData = {text: ''};

function NewToDo({AddFn}) {
    const [formData, setFormData] = useState(initialFormData);
    const handleChange = (evt) => {
        evt.preventDefault();
        const {name, value} = evt.target;
        setFormData(formData => ({...formData, [name]:value}));
    }
    const handleSubmit = (evt) => {
        evt.preventDefault();
        AddFn(formData);
        setFormData(initialFormData);
    }
    return (
        <form onSubmit={handleSubmit} data-testid="form">
            <br/> 
            <label htmlFor='item'>New Item: </label>
            <input id='item' name='text' value={formData.text} onChange={handleChange}/>    
        </form>
    )
}

export default NewToDo;