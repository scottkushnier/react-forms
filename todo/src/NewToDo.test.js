import { render } from "@testing-library/react";
import NewToDo from "./NewToDo";

test("newtodo smoke test", function () {
  render(<NewToDo />)
});

test("matches snapshot", function() {
    const {asFragment} = render(<NewToDo />);
    expect(asFragment()).toMatchSnapshot();
  });
