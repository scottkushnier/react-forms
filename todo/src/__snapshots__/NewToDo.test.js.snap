// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <br />
    <label
      for="item"
    >
      New Item: 
    </label>
    <input
      id="item"
      name="text"
      value=""
    />
  </form>
</DocumentFragment>
`;
